pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -623,12 +622,14 @@ public void validateBundleOwnership(NamespaceBundle bundle, boolean authoritativ
      */
     protected void validateTopicOwnership(TopicName topicName, boolean authoritative) {
         try {
-            validateTopicOwnershipAsync(topicName, authoritative).join();
-        } catch (CompletionException ce) {
-            if (ce.getCause() instanceof WebApplicationException) {
-                throw (WebApplicationException) ce.getCause();
+            validateTopicOwnershipAsync(topicName, authoritative)
+                    .get(config().getMetadataStoreOperationTimeoutSeconds(), SECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException ex) {
+            Throwable realCause = FutureUtil.unwrapCompletionException(ex);
+            if (realCause instanceof WebApplicationException) {
+                throw (WebApplicationException) realCause;
             } else {
-                throw new RestException(ce.getCause());
+                throw new RestException(realCause);
             }
```

Comment:
You could also consider an approach to replace all `.join()` calls at once in the PulsarWebResource subclasses instead of sending PRs one by one.

Reply:
@lhotari  Yes, you are right, thanks for the suggestion.

I need to say sorry, I didn't look closely at the other duplicate code because I found this issue while looking at other features at the time.
So, I will solve this problem according to your suggestion.